src/components/interface/AudioPlay.jsx

```diff
@@ -3,24 +3,94 @@ import PropTypes from 'prop-types';
 import { withStyles } from '@material-ui/core/styles';
 import PlayCircleOutlineIcon from '@material-ui/icons/PlayCircleOutline';
 import styles from '../styles';
+import Button from '../interface/Button';
+import * as firebase from 'firebase';
+
 
 class AudioPlayer extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      disabled: false
+      audioFile: undefined,
+      metadataLoaded: false
     };
   }
 
-  // TODO: handle audio playing file
+  loadAudio = () => {
+    //const {audioFile} = this.state;
+    this.pathRef.getDownloadURL().then(function (url) {
+      var xhr = new XMLHttpRequest();
+      xhr.responseType = 'blob';
+      xhr.onload = async function (event) {
+        var blob = xhr.response;
+        this.setState({ audioFile: new Audio(url) });
+        this.state.audioFile.addEventListener('loadeddata', () => {
+          this.setState({ metadataLoaded: true});
+        })
+      }.bind(this);
+      xhr.open('GET', url);
+      xhr.send();
+    }.bind(this)).catch(function (error) {
+      console.log("ERROR. Firebase audio file failed to load.");
+    });;
+  }
+
+  loadMetadata = () => {
+    const { audioFile } = this.state;
+
+    audioFile.preload = "auto";
+    audioFile.load();
+    //this.setState({metadataLoaded: true});
+  }
+
+  componentDidMount() {
+    const { audioURL } = this.props;
+    const { audioFile } = this.state;
+
+    console.log("HERE");
+
+    this.pathRef = firebase.storage().ref().child(audioURL);
+    if (audioFile == undefined) {
+      this.loadAudio();
+    }
+  }
+
+  componentWillUnmount() {
+    const { audioFile } = this.state;
+    if (audioFile != undefined) {
+      audioFile.pause();
+    }
+  }
+
+  handleClick = () => {
+    const { audioFile } = this.state;
+    if (audioFile.paused) {
+      audioFile.play();
+    }
+    else {
+      audioFile.pause();
+    }
+  }
 
+  formattedAudioDuration = () => {
+    const { audioFile } = this.state;
 
-  // TODO: create method to convert time to XX:XX format
+    let secs = audioFile.duration;
+    let formattedTime = Math.floor(secs / 60) + ":" + Math.floor(secs % 60);
```

Comment:
These can all be `const`s. General React styling is we use const unless we need to be able to alter the object.

If I recall correctly, you could even use a const and still manipulate the object. `const` binds the reference rather than the value.

Reply:
Got it, thanks